server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -309,6 +311,14 @@ public void createIndex(
         final CreateIndexClusterStateUpdateRequest request,
         final ActionListener<CreateIndexClusterStateUpdateResponse> listener
     ) {
+        // INDEX_STORE_TYPE_SETTING is intended to be a system-managed index setting that is configured when restoring a snapshot and
+        // should not be set to value REMOTE_SNAPSHOT by user.
+        String storeTypeSetting = request.settings().get(INDEX_STORE_TYPE_SETTING.getKey());
+        if (storeTypeSetting != null && storeTypeSetting.equals(RestoreSnapshotRequest.StorageType.REMOTE_SNAPSHOT.toString())) {
```

Comment:
This validation should probably go in the [`aggregateIndexSettings()` method](https://github.com/opensearch-project/OpenSearch/blob/6a6ab32e532d16de222645ccfae376c778c68991/server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java#L792-L793)

Reply:
I think you are right, this can go to aggregateIndexSettings() method. Will update in next commit. Even if we move this to aggregateIndexSettings() method, restore path wouldn't hit this before failing.